### PR TITLE
8 update map markers based on filters

### PR DESCRIPTION
**Pull request by cmhnk**
Addresses Issue #8 

The `Map` component was rendering the markers in `componentDidMount()`, which meant that even when the `points` props changed, the markers weren't being re-rendered. I moved the marker rendering into its own method, `updateMarkers()`, which gets called in `componentDidUpdate()`. 
Also, marker popups get cleared whenever the filters are adjusted.

### CODE REVIEW

File: rails/app/javascript/components/Map.jsx [hunk 1]
```diff
@@ -33,35 +21,57 @@ export default class Map extends Component {
...
+       .setLngLat(marker.geometry.coordinates)
+       .setPopup(popup) // sets a popup on this marker
+       .addTo(this.map);
+
+       el.addEventListener('click', () =>
+         {
+           this.props.onMapPointClick(marker.properties.stories);
+           this.map.panTo(marker.geometry.coordinates);
+         }
+       )
```
Comment (baweaver) [inline rails/app/javascript/components/Map.jsx]: I wonder if we should make this into a component

File: rails/app/javascript/components/Map.jsx [hunk 2]
```diff
@@ -33,35 +21,57 @@ export default class Map extends Component {
...
+       el.className = 'marker';
+       el.id = 'storypoint' + marker.id;
+       var popup = new mapboxgl
+         .Popup({ offset: 15 })
+         .setHTML('<h1>' + marker.properties.title + '</h1>' + '<h2>' + marker.properties.region + '</h2>')
+       // make a marker for each feature and add to the map
+       new mapboxgl.Marker(el)
+       .setLngLat(marker.geometry.coordinates)
+       .setPopup(popup) // sets a popup on this marker
+       .addTo(this.map);
```
Comment (baweaver) [inline rails/app/javascript/components/Map.jsx]: Nit: indentation

File: rails/app/javascript/components/Map.jsx [hunk 3]
```diff
@@ -33,35 +21,57 @@ export default class Map extends Component {
...
-           {
-             this.props.onMapPointClick(marker.properties.stories);
-             this.map.panTo(marker.geometry.coordinates);
-           }
-         )
-        });
-      });
+  updateMarkers() {
+    this.props.points.features.forEach(marker => {
+      (marker.properties);
```
Comment (baweaver) [inline rails/app/javascript/components/Map.jsx]: Old log line?

File: rails/app/javascript/components/Map.jsx [hunk 4]
```diff
@@ -33,35 +21,57 @@ export default class Map extends Component {
...
+       // create a HTML element for each feature
+       var el = document.createElement('div');
+       el.className = 'marker';
+       el.id = 'storypoint' + marker.id;
```
Comment (baweaver) [inline rails/app/javascript/components/Map.jsx]: Nit: interpolation could be used here:

```javascript
el.id = `storypoint${marker.id}`;
```

File: rails/app/javascript/components/Map.jsx (hunk 1 above)
Comment (baweaver) [inline rails/app/javascript/components/Map.jsx]: Nit: might update syntax a bit to make it consistent:

```javascript
el.addEventListener('click', () => {
  this.props.onMapPointClick(marker.properties.stories);
  this.map.panTo(marker.geometry.coordinates);
});
```

**Comment by cmhnk** [comment]
All the comments were actually on code that I just moved rather than wrote. Let's address in a separate PR!

**Pull request merged**